### ISSUE TITLE
Allow initial library to be unindexed

### ISSUE THREAD

**Issue by xogeny** [xogeny/impact#54]
It should be possible, as with `pip`, to install something even if it hasn't been indexed by `impact`, _e.g.,_

`impact install -e git+https://github.com/xogeny/Sensors.git#/path/to/library`

The main application for this, that I see, is the ability to load private libraries on GitHub, but I suspect there will be others.  Note, this only works for the initial library to be downloaded.  Any other libraries have to be indexed (or else `impact` can't know about them).

If other private libraries are required, then I see two possible options.  First, the user could create a private index file (so that `impact` could find and load additional dependencies).  Second, the user could list additional libraries in the `install` command as a way to preemptively load dependencies that are private.

**Comment by xogeny** [comment]
I'm fine with this option, but I think we should put it in a post-conference milestone.  Comments?

**Comment by dietmarw** [comment]
I agree that for the conference we should concentrate to have something user oriented in place (i.e., making **released** libraries installable). All the developer nice-to-haves should be pushed to after the conference at least.
